### PR TITLE
fix(compiler): cant call fromJson on imported structs

### PR DESCRIPTION
**Pull request by hasanaburayyan**
Closes: https://github.com/winglang/wing/issues/3792

Now supports calling fromJson on structs that were brought from other wing files

example: 
file_1.w
```js
struct Person {
	name: str;
}
```

file_2.w
```js
bring "./file1.w" as f1;

let j = { name: "cool" };
f1.Person.fromJson(j);
```

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/tests/valid/struct_from_json.w [hunk 1]
```diff
@@ -236,4 +236,16 @@ test "lifting a student" {
   assert(studentInflight1.dob.month == 10);
   assert(studentInflight1.dob.day == 10);
   assert(studentInflight1.dob.year == 2005);
-}
\ No newline at end of file
+}
+
+// bring structs from other files
+bring "./subdir/structs.w" as externalStructs;
```
Comment (eladb) [inline examples/tests/valid/struct_from_json.w:242]: Does this also work for structs imported via JSII? Maybe add a test..,

File: libs/wingc/src/type_check.rs [hunk 2]
```diff
@@ -4593,8 +4593,6 @@ impl<'a> TypeChecker<'a> {
 								}
 							}
 						}
-						let lookup = env.lookup(&s.name, None);
-						let type_ = lookup.unwrap().as_type().unwrap();
```
Comment (MarkMcCulloh) [inline libs/wingc/src/type_check.rs:4597]: nice catch

File: examples/tests/valid/struct_from_json.w (hunk 1 above)
Comment (MarkMcCulloh) [reply examples/tests/valid/struct_from_json.w:242]: Based on this https://github.com/winglang/wing/issues/3943 I take it does not and is being left for later

**Review by MarkMcCulloh — APPROVED** [review]

File: examples/tests/valid/struct_from_json.w (hunk 1 above)
Comment (hasanaburayyan) [reply examples/tests/valid/struct_from_json.w:242]: Yupp thats exactly right @MarkMcCulloh, Im brainstorming the JSII imported structs now. The issue here is that we arent jsifying those so we never create a jsonschema for them to use for validation. I have a few ideas that are probably not gonna work but may 😁

**Pull request merged**